parser-typechecker/src/Unison/Server/Backend.hs

```diff
@@ -195,12 +204,23 @@ findShallowInBranch codebase b = do
       b0 = Branch.head b
   termEntries <- for (R.toList . Star3.d1 $ Branch._terms b0) $ \(r, ns) -> do
     ot <- lift $ loadReferentType codebase r
-    pure $ ShallowTermEntry r (hqTerm b0 ns r) ot
+    let isDoc = case ot of
+          Just t  -> Typechecker.isSubtype t $ Type.ref mempty Decls.docRef
+          Nothing -> False
+        isTest =
+          Metadata.hasMetadataWithType r (Decls.isTestRef) $ Branch._terms b0
+        tag = if isDoc then Just Doc else if isTest then Just Test else Nothing
```

Comment:
What do you think of having the non-doc, non-test case be "Term" or similar instead dealing with the Maybe?

Reply:
I can see this evolving to a list instead of a Maybe later on, or something more complex. Inclined to leave it alone for now.